app/main/posts/common/post-actions.directive.js

```diff
@@ -44,14 +44,14 @@ function PostActionsDirective(
 
         function deletePost() {
             PostActionsService.delete($scope.post).then(function () {
-                // If we're not already on some top level view
-                if ($location.path().indexOf('views') === -1 &&
-                    $location.path().indexOf('collections') === -1 &&
-                    $location.path().indexOf('savedsearches') === -1) {
+                $rootScope.$broadcast('event:edit:post:status:data:mode:saveSuccess', {post: $scope.post, deleted: true});
+                // If we're not already on some of the posts views
+                if (!$state.includes('posts')) {
```

Comment:
it it's not a post.* view IDK how you even deleted a post, but we're taking you to the posts.data page anyway.

Reply:
Yup. This logic is a little backwards but much easier to parse not at least :)